search-commons/src/test/java/no/unit/nva/search/ElasticsearchTest.java

```diff
@@ -47,41 +49,83 @@ public class ElasticsearchTest {
     public static final String STATUS_TO_INCLUDE_IN_RESULT = "Pending";
     public static final int ZERO_HITS_BECAUSE_APPROVED_WAS_FILTERED_OUT = 0;
     public static final long DELAY_AFTER_INDEXING = 1000L;
-    private static final String ELASTICSEARCH_VERSION = "7.10.2";
     private static final int PAGE_SIZE = 10;
     private static final int PAGE_NO = 0;
     private static final URI ORGANIZATION_ID_URI_HARDCODED_IN_SAMPLE_FILES = URI.create("https://www.example.com/20754.0.0.0");
     private static final String COMPLETED = "Completed";
-    
-    @Container
-    public ElasticsearchContainer container = new ElasticsearchContainer(DockerImageName
-                                                                             .parse(ELASTICSEARCH_OSS)
-                                                                             .withTag(ELASTICSEARCH_VERSION));
+
     private SearchClient searchClient;
     private IndexingClient indexingClient;
 
+
+    @SuppressWarnings("rawtypes")
+    private final GenericContainer container = new GenericContainer(
+            DockerImageName.parse("opensearchproject/opensearch:latest"))
+            .withEnv("plugins.security.disabled", "true")
+            .withEnv("discovery.type", "single-node");
+
+    @SuppressWarnings("unchecked")
     @BeforeEach
     void setUp() {
-        RestClientBuilder restClientBuilder = RestClient.builder(HttpHost.create(container.getHttpHostAddress()));
+
+        container.setPortBindings(List.of("9200:9200", "9600:9600"));
+        container.setExposedPorts(List.of(9200, 9600));
+        container.start();
+
+        var port9200 = container.getMappedPort(9200);
+        var host = container.getHost();
+        var httpHostAddress = host + ":" + port9200;
```

Comment:
port9200 should be renamed to just port, or a name exaplining what the value means - not what it is. 
I see you also do the same thing in canConnectToContainer() - perhaps you can extract both to a function? like:
`getContainerAddress(container)`